flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/batch/table/CalcTest.scala

```diff
@@ -117,6 +120,45 @@ class CalcTest extends TableTestBase {
     util.verifyTable(resultTable, expected)
   }
 
+  @Test
+  def testSelectLiterals(): Unit = {
+    val util = batchTestUtil()
+    val sourceTable = util.addTable[Int]("MyTable", 'a)
+    val resultTable = sourceTable.select("ABC", BigDecimal(1234), LocalDateTime.of(1, 1, 1, 1, 1))
```

Comment:
Could you also verify that `LocalDateTime` fields still throw an exception when they are not a literal but a table field?